src/System.Diagnostics.EventLog/tests/Helpers.cs

```diff
@@ -60,10 +60,17 @@ public static T RetryOnAllPlatforms<T>(Func<T> func)
         public static void WaitForEventLog(EventLog eventLog, int entriesExpected)
         {
             int tries = 0;
-            while (RetryOnAllPlatforms((() => eventLog.Entries.Count)) < entriesExpected && tries < 20)
+            while (RetryOnAllPlatforms((() => eventLog.Entries.Count)) < entriesExpected && tries < 51)
             {
-                Thread.Sleep(100);
-                tries++;
+                if (tries == 50)
+                {
+                    Thread.Sleep(30000);
+                }
+                else
+                {
+                    Thread.Sleep(100 * (tries));
```

Comment:
Each call to RetryOnAllPlatforms is already doing 20 tries over a course of 2 seconds, so in the case where this is going to fail, that will end up taking ~100 seconds for the 51 retries at this level.  Then on top of that the last retry is 30 seconds, and then this 100*tries in between is a total of another ~130 seconds.  So this test may end up doing retries and waiting for over 4 minutes?

Reply:
RetryOnAllPlatforms is doing tries to get the count value and this getting count function rarely requires a retry.
The second part is to check whether count has been updated to required value. So it will be around 130 seconds i.e about 2 minutes. I can reduce that if there is some maximum time limit

Reply:
My reasoning was that if we put in long delays, we will find out whether the fragility is because we're not waiting long enough, or because the event is never going to happen. If we still get failures, we can reduce the sleeps down because they're not helping. If we no longer get failures, but long sleeps are required, we can mark tests [OuterLoop].

Reply:
We could possibly log (Console.Write) how long we ended up waiting, if it was more than a second or two.

Reply:
Please use Stopwatch.StartNew/Stopwatch.Elapsed to write to the console how long was waited if the wait time is greater than say 5 seconds.
Assuming these new sleeps solve the fragility, we can use these console messages to figure out how much to back off the sleeps.

Reply:
@danmosemsft, it failed again:
https://github.com/dotnet/corefx/issues/24874#issuecomment-343753418